core/src/main/java/hivemall/fm/FactorizationMachineUDTF.java

```diff
@@ -379,23 +379,28 @@ protected void checkInputVector(@Nonnull final Feature[] x) throws HiveException
         _model.check(x);
     }
 
+    protected void processValidationSample(@Nonnull final Feature[] x, final double y)
+            throws HiveException {
+        if (_adaptiveRegularization) {
+            trainLambda(x, y); // adaptive regularization
```

Comment:
`FFM fully ignores adaptive regularization option` is expected behavior.
Not tested AdaptiveRegularization with FFM and/or FTRL.

Reply:
373144d5151d1a57c3c13059ac70cc2e2908dc44